src/test/java/core/basesyntax/service/RegistrationServiceTest.java

```diff
@@ -0,0 +1,123 @@
+    private static final String INVALID_BORDERLINE_LOGIN = "user1";
+    private static final String INVALID_BORDERLINE_PASSWORD = "12345";
+    private static final int VALID_AGE = 18;
+    private static final int INVALID_BORDERLINE_AGE = 17;
+    private static final int INVALID_NEGATIVE_AGE = -2;
+    private RegistrationServiceImpl registrationService;
+    private StorageDaoImpl storageDao;
+
+    @BeforeEach
+    void setUp() {
+        registrationService = new RegistrationServiceImpl();
+        storageDao = new StorageDaoImpl();
+    }
+
+    @Test
+    void register_checkUserExist_notOk() {
+        User registeredUser = new User();
+        registeredUser.setLogin(VALID_LOGIN);
+        registeredUser.setPassword(VALID_PASSWORD);
+        registeredUser.setAge(VALID_AGE);
+        storageDao.add(registeredUser);
+        User newUser = new User();
+        newUser.setLogin(VALID_LOGIN);
+        newUser.setPassword(VALID_PASSWORD);
+        newUser.setAge(VALID_AGE);
+        ValidationException exception = assertThrows(ValidationException.class, () -> {
+            registrationService.register(newUser);
+        });
+        assertEquals("User with your login has already exists.Try new login!",
+                exception.getMessage());
+    }
+
+    @Test
+    void register_successRegistration_Ok() {
+        User registeredUser = new User();
+        registeredUser.setLogin(VALID_LOGIN);
+        registeredUser.setPassword(VALID_PASSWORD);
+        registeredUser.setAge(VALID_AGE);
+        User actualUser = registrationService.register(registeredUser);
+        assertEquals(registeredUser, actualUser);
+    }
+
+    @Test
+    void register_nullLogin_notOk() {
+        User userNullLogin = new User();
+        userNullLogin.setLogin(null);
+        ValidationException exception = assertThrows(ValidationException.class, () -> {
+            registrationService.register(userNullLogin);
+        });
+        assertEquals("Login cannot be null or less than 6 characters!", exception.getMessage());
+    }
+
+    @Test
+    void register_shortLogin_notOk() {
+        User userShortLogin = new User();
+        userShortLogin.setLogin(INVALID_BORDERLINE_LOGIN);
+        ValidationException exception = assertThrows(ValidationException.class, () -> {
+            registrationService.register(userShortLogin);
+        });
+        assertEquals("Login cannot be null or less than 6 characters!", exception.getMessage());
+    }
+
+    @Test
+    void register_shortPassword_notOk() {
+        User userShortPassword = new User();
+        userShortPassword.setLogin(VALID_LOGIN);
+        userShortPassword.setPassword(INVALID_BORDERLINE_PASSWORD);
+        ValidationException exception = assertThrows(ValidationException.class, () -> {
+            registrationService.register(userShortPassword);
+        });
+        assertEquals("Password cannot be null or less than 6 characters!", exception.getMessage());
```

Comment:
The exception message should be specific to the failure, in this case, it should be about the password being too short.